instrumentation/kafka/src/main/java/brave/kafka/TracingConsumer.java

```diff
@@ -0,0 +1,181 @@
+package brave.kafka;
+
+import brave.Tracing;
+import org.apache.kafka.clients.consumer.*;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+
+public class TracingConsumer<K, V> implements Consumer<K, V> {
```

Comment:
make this package private and instead make a decorator similar to how some of the other libraries work.
This can help centralize config I think.

ex. 
```java
traced = kafkaTracing.consumer(consumer);
```

For example, KafkaTracing may have additional configuration, and this could be done in a builder vs ctor overloads here. Since we add no new methods, hiding the type by returning Consumer (and making ctor package private) is safest start!